src/shared/components/ResourceEditor/useEditorTooltip.tsx

```diff
@@ -0,0 +1,405 @@
+      }
+    }
+    function hideTooltip(tooltip: HTMLDivElement) {
+      if (!tooltip.parentNode) {
+        return;
+      }
+      setTimeout(() => {
+        removeTooltipFromDom(tooltip);
+      }, 300);
+    }
+    function showTooltip(content: HTMLDivElement, node: HTMLElement) {
+      const tooltip = document.createElement('div');
+      tooltip.className = CODEMIRROR_HOVER_CLASS;
+      tooltip.appendChild(content);
+      document.body.appendChild(tooltip);
+
+      function hide() {
+        if (tooltip) {
+          node.classList.remove('has-tooltip');
+          hideTooltip(tooltip);
+          tooltip.remove();
+        }
+        node.removeEventListener('mouseout', hide);
+        node.removeEventListener('click', hide);
+        node.removeEventListener('scroll', hide);
+      }
+
+      node.addEventListener('mouseout', hide);
+      node.addEventListener('click', hide);
+      node.addEventListener('scroll', hide);
+
+      const pool: ReturnType<typeof setTimeout> = setTimeout(() => {
+        if (tooltip) {
+          hideTooltip(tooltip);
+          tooltip.remove();
+        }
+        return clearTimeout(pool);
+      }, 3000);
+
+      return tooltip;
+    }
+
+    async function onMouseOver(ev: MouseEvent) {
+      const node = ev.target as HTMLElement;
+      if (node) {
+        const { token } = getTokenAndPosAt(ev, currentEditor);
+        const content = token?.string || '';
+        const url = content.replace(/\\/g, '').replace(/\"/g, '');
+        if (url && mayBeResolvableLink(url)) {
+          node.classList.add('wait-for-tooltip');
+          removeTooltipsFromDOM();
+          editorLinkResolutionHandler({
+            nexus,
+            apiEndpoint,
+            url,
+            orgLabel,
+            projectLabel,
+          }).then(({ resolvedAs, results, error }) => {
+            const tooltipContent = createTooltipContent({
+              resolvedAs,
+              error,
+              results,
+            });
+            if (tooltipContent) {
+              node.classList.remove('wait-for-tooltip');
+              node.classList.add(
+                resolvedAs === 'error'
+                  ? 'error'
+                  : resolvedAs === 'resource' &&
+                    (results as TDELink).isDownloadable
+                  ? 'downloadable'
+                  : 'has-tooltip'
+              );
+              const tooltip = showTooltip(tooltipContent, node);
+              const calculatePosition = (ev: MouseEvent) =>
+                positionner(ev, tooltip);
+              editorWrapper.addEventListener('mousemove', calculatePosition);
+            }
+          });
+        }
+      }
+    }
+    // allow the tooltip only when the editor is not in edition mode
+    // and the popover is not open
+    allowTooltip && editorWrapper.addEventListener('mouseover', onMouseOver);
+    // remove the event listener when not allwoed
+    !allowTooltip &&
```

Comment:
Hmm why this? In the previous line aren't we only adding the listener when the allowTip is true? If so, there's no need to remove it, since it was never added.

Reply:
This one is to remove it, there is two cases where we have to remove the listener
1. The component is un-mounting  
2. The editor is in the editing mode, then hover event must be removed

so this one is the one that remove the event listener when the user start editing
and the one in the cleanup function for the un-mounting